### PR TITLE
Compress and fingerprint javascript assets

### PR DESCRIPTION
**Pull request by cantsin**
[Link to ZenHub issue.](https://app.zenhub.com/workspaces/doj-crt-intake-scrum-board-5d03bf56c1c8a35d482eca0f/issues/18f/crt-portal/792)

## What does this change?

Turns on compressor (bundles javascript and css assets), and fingerprints them via ManifestFiles.

Bundling is on for all environments and provides a modest speedup when loading the page. 

I tried my best to get rid of gulp but was unable to do so, mostly because of the way USDWS is set up. I have, however, cleaned up the `gulpfile.js` and removed some stale code/libraries.

---

*Note*: Testing fingerprinting on S3 is difficult unless you already have localstack set up. Here's an alternative to test fingerprinting: put this line in local settings:

```python
COMPRESS_ENABLED = True
```

You should see something like:

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [x] Check for [accessibility](/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [x] Re-check for [accessibility](/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/crt_portal/settings.py [hunk 1]
```diff
@@ -336,9 +353,6 @@
     ('css', 'compressor_toolkit.precompilers.SCSSCompiler'),
 )
 
-# would like to add this before public release
-COMPRESS_ENABLED = False
```
Comment (cantsin) [inline crt_portal/crt_portal/settings.py:340]: By default this is set to whatever DEBUG is, so omitting this line for convenience

File: gulpfile.js [hunk 2]
```diff
@@ -81,10 +79,6 @@ gulp.task('build-sass', function(done) {
   var plugins = [
     // Autoprefix
     autoprefixer(autoprefixerOptions),
-    // Pack media queries
-    mqpacker({ sort: true }),
-    // Minify
-    cssnano(({ autoprefixer: { browsers: autoprefixerOptions }}))
```
Comment (cantsin) [inline gulpfile.js:87]: css-mqpacker is no longer supported -- see https://www.npmjs.com/package/css-mqpacker

cssnano is no longer needed thanks to django compressor. there _shouldn't_ be any difference between cssnano/django-compressor, and I can't see any difference in the resulting design.

File: crt_portal/crt_portal/settings.py (hunk 1 above)
Comment (Jkrzy) [reply crt_portal/crt_portal/settings.py:340]: 👍 , Noting ref from docs: The default value for `COMPRESS_ENABLED` is the _opposite_ of `DEBUG`

https://django-compressor.readthedocs.io/en/stable/settings/#django.conf.settings.COMPRESS_ENABLED

File: crt_portal/cts_forms/templates/forms/base.html [hunk 3]
```diff
@@ -99,8 +104,10 @@
     </div>
     {% endblock %}
   </body>
+  {% compress js %}
   <script src="{% static 'js/url_params_polyfill.js' %}"></script>
   <script src="{% static 'js/uswds.min.js' %}"></script>
   <script src="{% static 'js/focus_alert.js' %}"></script>
   {% block page_js %}{% endblock %}
```
Comment (Jkrzy) [inline crt_portal/cts_forms/templates/forms/base.html:111]: We've got some external sources in templates that use this `page_js` block that end up leading to an exception when testing locally -- touchpoints script in `report_primary_complaint.html` and `confirmation.html` specifically

```
UncompressableFileError at /report/
'https://touchpoints.app.cloud.gov/touchpoints/e0517981/js' isn't accessible via COMPRESS_URL ('/static/') and can't be compressed
```

**Review by Jkrzy — CHANGES_REQUESTED** [review]
Thanks @cantsin! In testing locally this breaks report submission, left one comment in-line where I think the issue lies.

File: crt_portal/cts_forms/templates/forms/base.html (hunk 3 above)
Comment (cantsin) [reply crt_portal/cts_forms/templates/forms/base.html:111]: Gosh, great catch, thank you!!

Comment (cantsin) [reply crt_portal/cts_forms/templates/forms/base.html:111]: Fixed + rebased on dev

File: crt_portal/cts_forms/templates/forms/confirmation.html [hunk 4]
```diff
@@ -189,7 +189,10 @@ <h2 class='h1__display'>{% trans "Your submission" %}</h2>
 {% endblock footer_extra %}
 
 {% block page_js %}
-  <script src="https://touchpoints.app.cloud.gov/touchpoints/a24aeb5e/js" async></script>
   <script src="{% static 'js/modal.js' %}"></script>
   <script src="{% static 'js/redirect-modal.js' %}"></script>
 {% endblock %}
+
+{% block touchpoint_js %}
```
Comment (Jkrzy) [inline crt_portal/cts_forms/templates/forms/confirmation.html:196]: Looks like the implementation in `report_primary_complaint.html` here differs by one character so it doesn't extend this block. 
```suggestion
{% block touchpoints_js %}
```

File: crt_portal/cts_forms/templates/forms/base.html [hunk 5]
```diff
@@ -99,8 +104,13 @@
...
+  {% endcompress %}
+
+  {# do not compress #}
+  {% block touchpoint_js %}{% endblock %}
```
Comment (Jkrzy) [inline crt_portal/cts_forms/templates/forms/base.html:115]:
```suggestion
  {% block touchpoints_js %}{% endblock %}
```

**Review by Jkrzy — CHANGES_REQUESTED** [review]

File: crt_portal/cts_forms/templates/forms/base.html (hunk 5 above)
Comment (cantsin) [reply crt_portal/cts_forms/templates/forms/base.html]: Oh gosh, egg on my face. Thanks for the catch!!

**Review by Jkrzy — APPROVED** [review]

**Comment by Jkrzy** [comment]
@cantsin, this one failed to deploy to dev -- can you take a look?

**Pull request merged**